dist/tools/doccheck/check.sh

```diff
@@ -1,22 +1,73 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 # Copyright 2014 Oliver Hahm <oliver.hahm@inria.fr>
+#           2018 Kaspar Schleiser <kaspar@schleiser.de>
+#           2018 Alexandre Abadie <alexandre.abadie@inria.fr>
 #
 # This file is subject to the terms and conditions of the GNU Lesser
 # General Public License v2.1. See the file LICENSE in the top level
 # directory for more details.
 
 RIOTBASE=$(readlink -f "$(dirname $(realpath $0))/../../..")
 
+if tput colors &> /dev/null && [ $(tput colors) -ge 8 ]; then
+    CERROR="\e[1;31m"
+    CWARN="\033[1;33m"
+    CRESET="\e[0m"
+else
+    CERROR=
+    CWARN=
+    CRESET=
+fi
+
 ERRORS=$(make -C "${RIOTBASE}" doc 2>&1 | \
             grep '.*warning' | \
             sed "s#${PWD}/\([^:]*\)#\1#g")
 
 if [ -n "${ERRORS}" ]
 then
-    echo "ERROR: Doxygen generates the following warnings:"
+    echo -e "${CERROR}ERROR: Doxygen generates the following warnings:${CRESET}"
     echo "${ERRORS}"
     exit 2
-else
-    exit 0
+fi
+
+# Check all groups are defined
+DEFINED_GROUPS=$(git grep @defgroup -- '*.h' '*.txt' | \
+                  grep -v vendor | \
+                  grep -v examples | \
+                  grep -v tests | \
```

Comment:
Instead of all these line repetitions do:

```sh
exclude_filter() {
    grep -v -e vendor -e examples -e tests
}
```

And then use it like this in this instance

```sh
DEFINED_GROUPS=$(git grep @defgroup -- '*.h' '*.txt' | \
                  exclude_filter | \
                  grep -oE '@defgroup[ ]+[^ ]+' | \
                  grep -oE '[^ ]+$' | sort -u)
```

also below of course